_posts/2020-08-17-creating-your-ci-pipeline-with-concource.md

```diff
@@ -0,0 +1,217 @@
+and what is not). Anyway, in context of this blog post we can just think that we have *things* where we get 
+something (input) and *things* where we put something (output). These can be external *things*, or 
+internal *things* to pass some result between tasks.
+
+There is some Concourse core resources available which you can use without defining resource type explicitly
+and [there is also some third-party resources listed on Concourse website](https://resource-types.concourse-ci.org/).
+In addition, there is also many other in the wild Internet.
+
+However, **keep security concerns in mind** whenever you use third-party resource types and especially
+with the ones which are not listed on Concource website. Resource types are in practice just Docker images, so
+if you are unlucky you can find yourself pulling an image which is changed to something nasty without you knowing it.
+Fortunately, it's quite an easy job to make your own resource type. So, if you feel like
+you don't trust enough for the publisher of some resource type, you can use it as base for your own resource type.
+However, remember to contribute to the base project if you make good improvements or bug fixes. 
+
+Let's build and deploy some dummy docker images into the Docker Hub 
+------------------
+
+I don't go into details how to start and run Concourse, but I have provided some instructions
+in [my example project in Github](https://github.com/solita-alperttiti/concource-ci-example).
+Also, fully runnable pipeline example can be found from that project.
+
+Every pipeline in Concource has its main pipeline definition file. So, it's good to start by creating one.
+In my case it's called *concource-example.pipeline.yml*, but you can name it whatever you like.
+Because every pipeline has least one input, let's start with it.
+
+```yaml
+- name: concource-example
+  type: git
+  source:
+    uri: ((git-source))
+    branch: ((git-branch))
+```
+
+So, now we have our input. Its type is *git*, which is one of core types of Concource, so you
+don't have to define explicitly its resource type in your definition. This syntax is quite easy to understand,
+it's an abstraction of some Git-branch in some repository. This example snippet doesn't tell
+much about which branch and which repository we are using, though.
+
+You can use exact path to your repository in pipeline definition if you wish, but I have used
+variables. My main goal has been to create pipeline definition which is more like a template for
+set of pipelines than just one pipeline for specific use. In this way you can easily create multiple
+pipelines from same template and just provide different variables for each. This is especially
+useful if you wish to use it with artifacts which have common steps. One real world
+example could be software with microservice architecture pattern. With this approach you have a shared pipeline definition
+for all of your services.
```

Comment:
a set of pipelines
for a specific use
In this way, 
from the same template
which have -> that have
real-world
a microservice
With this approach,